cmd/nerdctl/update.go

```diff
@@ -333,15 +337,21 @@ func updateContainer(ctx context.Context, client *containerd.Client, id string,
 
 	if err := updateContainerSpec(ctx, container, spec); err != nil {
 		log.G(ctx).WithError(err).Errorf("Failed to update spec %+v for container %q", spec, id)
+		// reset spec on error.
+		if err := updateContainerSpec(ctx, container, oldSpec); err != nil {
+			log.G(ctx).WithError(err).Errorf("Failed to update spec %+v for container %q", oldSpec, id)
+		}
 	}
```

Comment:
not sure if it's necessary to `defer` the `reset to old spec` logic. And the `retErr` is not used anywhere. So change to normal execution.